modules/tensor_mechanics/src/kernels/ALEKernel.C

```diff
@@ -0,0 +1,37 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ALEKernel.h"
+
+template<>
+InputParameters validParams<ALEKernel>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Sets up derivetives with respect to initial configuration");
+  return params;
+}
+
+ALEKernel::ALEKernel(const InputParameters & parameters) :
+  Kernel(parameters),
+  _assembly_undisplaced(_fe_problem.assembly(_tid)),
+  _grad_phi_undisplaced(_assembly_undisplaced.gradPhi())
+{
+}
+
+void
+ALEKernel::computeJacobian()
+{
+  _fe_problem.prepareShapes(_var.number(), _tid);
```

Comment:
Oh, let me think about this. I guess with use_displaced_mesh the shapes are
only prepared for the displaced problem, and _fe_problem is the
*undisplaced* problem. So it seems to me like the call is correct!

On Wed, Jun 22, 2016 at 4:39 PM Pritam Chakraborty <notifications@github.com>
wrote:

> In modules/tensor_mechanics/src/kernels/ALEKernel.C
> <https://github.com/idaholab/moose/pull/7228#discussion_r68064969>:
>
> > +  InputParameters params = validParams<Kernel>();
> > +  params.addClassDescription("Sets up derivetives with respect to initial configuration");
> > +  return params;
> > +}
> > +
> > +ALEKernel::ALEKernel(const InputParameters & parameters) :
> > +  Kernel(parameters),
> > +  _assembly_undisplaced(_fe_problem.assembly(_tid)),
> > +  _grad_phi_undisplaced(_assembly_undisplaced.gradPhi())
> > +{
> > +}
> > +
> > +void
> > +ALEKernel::computeJacobian()
> > +{
> > +  _fe_problem.prepareShapes(_var.number(), _tid);
>
> I have moved the prepareShapes from StressDivergence to here. Right now
> this class just sets up the shapes using undisplaced mesh. Should I remove
> this class altogether and make this call from StressDivergence?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7228/files/e87894dae64c7f83d836fe474990dae447f50c9f#r68064969>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMWPsivHSWZSF9Ht7qfMcyp6RLXz5K4ks5qOUkYgaJpZM4I7uhF>
> .
>